examples/simulate_planar_pcs.py

```diff
@@ -13,233 +13,795 @@
+        if bool_plot:
+            plt.show()
+        plt.close()
+    
+    # ====================================================
+    # First evaluation of the potential energy to trigger JIT compilation
+    print("Evaluating the potential energy for the first time (JIT-compilation)...")
+    
+    timer_start = time.time()
+    U_ts = vmap(partial(auxiliary_fns["potential_energy_fn"], robot_params))(q_ts).block_until_ready()
+    U_ts = U_ts.block_until_ready()  # ensure the potential energy is computed
+    timer_end = time.time()
+    
+    simulation_dict["execution_time"]["compile_potential_energy"] = timer_end - timer_start
+    
+    print(f"Evaluating the potential energy took {timer_end - timer_start:.2e} seconds. \n")
+    
+    # Second evaluation of the potential energy to capture the time of the evaluation
+    print("Evaluating the potential energy for a second time (JIT-evaluation)...")
+    
+    timer_start = time.time()
+    U_ts = vmap(partial(auxiliary_fns["potential_energy_fn"], robot_params))(q_ts).block_until_ready()
+    U_ts = U_ts.block_until_ready()  # ensure the potential energy is computed
+    timer_end = time.time()
+    
+    simulation_dict["execution_time"]["evaluate_potential_energy"] = timer_end - timer_start
+    
+    simulation_dict["results"]["U_ts"] = U_ts
+    
+    print(f"Evaluating the potential energy for a second time took {timer_end - timer_start:.2e} seconds. \n")
+    
+    # ====================================================
+    # First evaluation of the kinetic energy to trigger JIT compilation
+    print("Evaluating the kinetic energy for the first time (JIT-compilation)...")
+    
+    timer_start = time.time()
+    T_ts = vmap(partial(auxiliary_fns["kinetic_energy_fn"], robot_params))(q_ts, q_d_ts).block_until_ready()
+    T_ts = T_ts.block_until_ready()  # ensure the kinetic energy is computed
+    timer_end = time.time()
+    
+    simulation_dict["execution_time"]["compile_kinetic_energy"] = timer_end - timer_start
+    
+    print(f"Evaluating the kinetic energy took {timer_end - timer_start:.2e} seconds. \n")
+    
+    # Second evaluation of the kinetic energy to capture the time of the evaluation
+    print("Evaluating the kinetic energy for a second time (JIT-evaluation)...")
+      
+    timer_start = time.time()
+    T_ts = vmap(partial(auxiliary_fns["kinetic_energy_fn"], robot_params))(q_ts, q_d_ts).block_until_ready()
+    T_ts = T_ts.block_until_ready()  # ensure the kinetic energy is computed
+    timer_end = time.time()
+    
+    simulation_dict["execution_time"]["evaluate_kinetic_energy"] = timer_end - timer_start
+    
+    simulation_dict["results"]["T_ts"] = T_ts
+    
+    print(f"Evaluating the kinetic energy for a second time took {timer_end - timer_start:.2e} seconds. \n")
+    
+    if bool_print:
+        print("U_ts =\n", U_ts)
+        print("T_ts =\n", T_ts)
+    
+    if bool_plot or bool_save_plot:
+        print("Plotting the energy... \n")
```

Comment:
Can you please put this code into a separate file and keep the original `simulate_planar_pcs.py` file unchanged? You could name the new file, for example, `examples/benchmark_planar_pcs_num.py`